src/Bicep.LangServer/Handlers/BicepDefinitionHandler.cs

```diff
@@ -472,6 +451,50 @@ copyVariablesDeclaration.Value is JArray copyVariablesArray &&
             return new();
         }
 
+        private LocationOrLocationLinks? TryHandleImportedSymbolLocationInBicepSource(Range originSelectionRange, CompilationContext context, ISemanticModel sourceModel, string? originalSymbolName, IArtifactReferenceSyntax enclosingDeclaration)
+        {
+            Uri? externalSourceUri = null;
+            SemanticModel? bicepModel = sourceModel as SemanticModel; // if the source model is a local Bicep file
+
+            if (sourceModel is ArmTemplateSemanticModel &&
+                moduleDispatcher.TryGetArtifactReference(context.Compilation.SourceFileGrouping.EntryPoint, enclosingDeclaration).IsSuccess(out var artifactReference)
+                && artifactReference is OciArtifactReference ociArtifactReference
+                && ociArtifactReference.TryLoadSourceArchive().TryUnwrap() is SourceArchive sourceArchive)
+            {
+                // An imported remote artifact was published with source, so we want to take the user to the definition in the source archive.
+                // Long term, it would be best to package up symbol information with the source that we could query here to find the location of the definition in the source file.
+                // But for this purpose, doing a compilation of the source entrypoint file should be enough to find the definition for most scenarios, even if the source file
+                // does not compile error-free (which is likely). We will compile the entrypoint file only, not any of its dependencies or referenced modules.
+                var explorer = new InMemoryFileExplorer.FileStore();
+                var compilationUri = new Uri("inmemory:///main.bicep");
+                var importedSourceBicep = sourceArchive.FindSourceFile(sourceArchive.EntrypointRelativePath).Contents;
+                var bicepFile = sourceFileFactory.CreateBicepFile(compilationUri, importedSourceBicep);
+
+                var workspace = new Workspace();
+                workspace.UpsertSourceFile(bicepFile);
+                var compilation = bicepCompiler.CreateCompilationWithoutRestore(compilationUri, workspace);
+
+                bicepModel = compilation.GetEntrypointSemanticModel();
+                externalSourceUri = BicepExternalSourceRequestHandler.GetRegistryModuleSourceLinkUri(ociArtifactReference, sourceArchive);
+            }
+
+            if (bicepModel?.Root.Declarations.Where(type => LanguageConstants.IdentifierComparer.Equals(type.Name, originalSymbolName)).FirstOrDefault() is { } originalDeclaration)
+            {
+                // entire span of the declaredSymbol
+                var targetRange = PositionHelper.GetNameRange(bicepModel.SourceFile.LineStarts, originalDeclaration.DeclaringSyntax);
```

Comment:
Since we're only interested in the span of the type symbol, I think we can just parse the file and locate the `TypeDeclarationSyntax` where the identifier name matches `originalSymbolName`. This eliminates the need for performing unnecessary semantic analysis on the linked source file.

Reply:
Good point, I've added this comment:

            // CONSIDER using only the syntax tree, not the semantic model

I'd rather not do this for now, though, mostly because that would change the code for the existing case of having a semantic model, which I know works.  Also, this is not a performance critical path, so I don't think the effort is worth it right now.